src/brogue/Utilities.c

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Utilities.c
+ *  Brogue
+ *
+ *  Copyright 2012. All rights reserved.
```

Comment:
fyi this header isn't a legal necessity, the license applies to all the source code, this is just a safeguard in case the file is somehow obtained on its own. It also declares an old copyright date with no rights holder, so is kind of meaningless in that aspect. It does no harm in staying, but just keep in mind we don't need the headers, if we started to care about copyright/licensing specifics we'd probably have to change the notices anyway

Reply:
Yeah I have some use for these general string functions. Also, you mentioned having thought about creating a utility file so I figured I'd go for it.

On the legal header, I just assumed each file needed one. That's about all the thought I put into it. Lemme know if you want me to yank it.

Reply:
It's fine to stay, does no harm I think.

The only other thing you might want to know is that `append` looks basically a home-made version of `strncat`. So you might prefer to use that as it's standard